Warn strata

jmhatch opened the pull request:
@andybeet Do you want to take a look at this PR, should close issue #59. The following was done:

- Included `lifecycle` in Imports / DESCRIPTION to use deprecated badges (via `usethis`)
- deprecated badges were added to documentation for datasets that are being deprecated (BTS, EcoMon, Shellfish, and Shrimp)
- Updated Roxygen version in DESCRIPTION
- Added an *.R file to document the pkg itself
- Added .onAttach() to issue a warning about data deprecation to the user when the pkg is attached

```r
> library(NEFSCspatial)
Note: The following datasets are deprecated and will be removed in a future version:
  • BTS_Strata
  • EcoMon_Strata
  • Shellfish_Strata
  • Shrimp_Strata
Please use https://mdeb-nefsc-noaa.hub.arcgis.com/ instead.
```

We can change the message once the mechanism for accessing the Strata data is fleshed out.

jmhatch commented:
I played around with `delayedAssign()`, but I couldn't quite get it to do what I wanted consistently. I'm just not that familiar with that function and how best to implement it for data deprecation. I thought issuing a message onAttach was the next best option.

andybeet commented:
ok, i'll take a look. thanks

andybeet reviewed (commented):
This looks good! Pkgdown site looks good too.
One question: Is it possible to add a link to the MDEB dataset for each of the deprecated files? Not a big deal really since i dont think we have many users. Just wondering.
Everything works as expected.

jmhatch commented:
@andybeet I added a @note to the documentation of each deprecated dataset with a link to the corresponding MDEB data. Good idea! If I place the link after the badge it messes up the Roxygen formatting, so I just made it a note.

Also changed the onAttach message, does the following make sense?
```r
## Issue deprecation warning 
.onAttach <- function(libname, pkgname) {
  packageStartupMessage(
    paste(
      "Note: The following datasets are deprecated and will be removed in a future version:",
      "  • BTS_Strata",
      "  • EcoMon_Strata",
      "  • Shellfish_Strata",
      "  • Shrimp_Strata",
      "Please use https://mdeb-nefsc-noaa.hub.arcgis.com/datasets/ instead:",
      "  • noaa::bottom-trawl-survey/about",
      "  • noaa::ecosystem-monitoring-survey/about",
      "  • noaa::atlantic-surfclam-and-ocean-quahog-survey/about",
      "  • noaa::sea-scallop-survey/about",
      "  • noaa::northern-shrimp-survey/about",
      sep = "\n"
    )
  )
}
```
And I'll merge with `dev` instead of `main`, didn't realize we had a `dev` branch. Thanks.

andybeet commented:
looks great! and, yes (in the onAttach msg), i understand that the extra string needs to be appended to url!. i'll merge this
